common-service-core/src/main/java/com/ca/mfaas/eurekaservice/client/util/EurekaMetadataParser.java

```diff
@@ -151,4 +151,41 @@ private RoutedService processUrls(Map<String, String> routeMap,
 
         return null;
     }
+
+    /**
+     * Generate Eureka metadata for ApiInfo configuration
+     *
+     * @param serviceId the identifier of a service which ApiInfo configuration belongs
+     * @return the generated Eureka metadata
+     */
+    public static Map<String, String> generateMetadata(String serviceId, ApiInfo apiInfo) {
+        Map<String, String> metadata = new HashMap<>();
+        String encodedGatewayUrl = UrlUtils.getEncodedUrl(apiInfo.getGatewayUrl());
+
+        if (apiInfo.getGatewayUrl() != null) {
+            metadata.put(String.format("%s.%s.%s", API_INFO, encodedGatewayUrl, API_INFO_GATEWAY_URL), apiInfo.getGatewayUrl());
+        }
+
+        if (apiInfo.getVersion() != null) {
+            metadata.put(String.format("%s.%s.%s", API_INFO, encodedGatewayUrl, API_INFO_VERSION), apiInfo.getVersion());
+        }
+
+        if (apiInfo.getSwaggerUrl() != null) {
+            UrlUtils.validateUrl(apiInfo.getSwaggerUrl(),
+                () -> String.format("The Swagger URL \"%s\" for service %s is not valid", apiInfo.getSwaggerUrl(), serviceId)
```

Comment:
Unhandled exception type MalformedURLException

Reply:
This was fixed. I can't see the change here in the PR, but I checked the file 
api-layer/common-service-core/src/main/java/com/ca/mfaas/eurekaservice/client/util/EurekaMetadataParser.java
directly with github in this branch (private/pza/enablers plain java config) and it is there.